File: app/src/main/java/com/konopko/stocktest/ViewModelTickerDetails.java
package com.konopko.stocktest;

import androidx.annotation.NonNull;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.konopko.stocktest.repository.IRepositoryTicker;

import javax.inject.Inject;

public class ViewModelTickerDetails extends ViewModel {

    @Inject
    IRepositoryTicker repositoryTicker;

    private MutableLiveData<Ticker> dataTicker = new MutableLiveData<>();

    public ViewModelTickerDetails(@NonNull String tickerId){
        App.getComponent().inject(this);
        dataTicker.setValue(repositoryTicker.getTicker(tickerId));
    }

    public LiveData<Ticker> getDataTicker(){
        return dataTicker;
    }

}

File: app/src/main/java/com/konopko/stocktest/ViewModelTickerDetails.kt
package com.konopko.stocktest

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import com.konopko.stocktest.repository.IRepositoryTicker
import org.koin.core.KoinComponent
import org.koin.core.inject

class ViewModelTickerDetails constructor() : ViewModel(), KoinComponent {

    private val repositoryTicker : IRepositoryTicker by inject()

    val dataTicker = MutableLiveData<Ticker>()

    constructor(tickerId: String): this(){
        dataTicker.value = repositoryTicker.getTicker(tickerId)
    }

}